src/components/Input/Input.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from "react";
+import { nanoid } from "nanoid";
+import Field from "../internal/Field";
+import InternalInput from "../internal/InternalInput";
+import { InternalInputProps } from "./types";
+
+const Input = (props: InternalInputProps) => {
+  const {
+    type = InternalInput.DEFAULT_PROPS.type,
+    variant = InternalInput.DEFAULT_PROPS.variant,
+    prefix,
+    suffix,
+    maxLength,
+    autoComplete,
+    label,
+    placeholder,
+    helpText,
+    innerRef,
+    disabled = false,
+    pasteAllowed = true,
+    optional = false,
+    onChange,
+    onBlur,
+    onFocus,
+    testId,
+    value,
+    error = [],
+    id,
+    __internal__focus,
+  } = props;
+
+  const [inputId] = useState(() => id ?? `input-${nanoid()}`);
+  const [auxId] = useState(() => `input-aux-${nanoid()}`);
+
+  const fieldErrors = Array.isArray(error) ? error : [error];
+  const hasErrors = Array.isArray(error) ? error.length !== 0 : !!error;
+
+  return (
+    <Field
+      optional={optional}
+      disabled={disabled}
+      label={label}
+      labelFor={inputId}
+      auxId={auxId}
+      helpText={helpText}
+      errors={fieldErrors}
+      testId={testId}
+    >
+      <InternalInput
+        id={label ? inputId : ""}
```

Comment:
Based on @strahinjaajvaz comment, can we do undefined instead of the empty string?
Let's make sure that `id` attribute is not added to the markup when `label` doesn't exist.